### PR TITLE
Added MassTell Mod

### PR DESCRIPTION
**Pull request by dgramop**
Supercedes #28 

Hello!

Thanks for reading over this! 
I just added a mod that individually PM's everyone on the server. Use case would be servers that block chat, but allow PM, or for making a broadcast that seems less like a broadcast. 

Why not /tell @a?
 ^- Many servers have @e @a @p etc. turned off. ;'(

Thanks,
Skype dhruv.gram
MC dgramop

### CODE REVIEW

**Comment by liach** [comment]
Compile error! Class name and file name!

**Comment by dgramop** [comment]
Duh. (facepalm)

On Wed, Jun 1, 2016 at 9:40 PM, liach notifications@github.com wrote:

> Compile error! Class name and file name!
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/Wurst-Imperium/Wurst-Client-for-MC-1.9.X/pull/30#issuecomment-223173773,
> or mute the thread
> https://github.com/notifications/unsubscribe/AKcTuHxGScWakKF9qre_A8K1jc5ERL3Cks5qHjSWgaJpZM4IsCuR
> .

## 

Regards,

## Dhruv Gramopadhye

Controll Freak Operator and Everything
dGRAMOP Development
Penetration Testing | Animation and Rendering | Web Development | Mobile

## App Development | Embedded Electronics | and Misc Java Development.

http://socnet.16mb.com - My most recent and best project - still buggy,
still work in progress. Feel free to create an account
http://dgramop.co - An index page of my projects
http://m.freeplay.url.ph - A neat looking version of the latter link
http://freeplay.url.ph - An index page of random experiments.
http://ppn.hol.es - A currency system that is online that I use to "bribe"
other children. Over 1.5 years old, I already can name a bunch of security
flaws. Try ../ attack when creating an account.
http://m.ppn.hol.es - An even more neat looking version of the above,
scrolling only works on mobile browsers unless you really zoom in. The junk
text is there to enable scrolling.

**Comment by dgramop** [comment]
I feel so stupid. I fixed it :/

On Wed, Jun 1, 2016 at 9:50 PM, D Gram dgramopadhye@gmail.com wrote:

> Duh. (facepalm)
> 
> On Wed, Jun 1, 2016 at 9:40 PM, liach notifications@github.com wrote:
> 
> > Compile error! Class name and file name!
> > 
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > https://github.com/Wurst-Imperium/Wurst-Client-for-MC-1.9.X/pull/30#issuecomment-223173773,
> > or mute the thread
> > https://github.com/notifications/unsubscribe/AKcTuHxGScWakKF9qre_A8K1jc5ERL3Cks5qHjSWgaJpZM4IsCuR
> > .
> 
> ## 
> 
> Regards,
> 
> ## Dhruv Gramopadhye
> 
> Controll Freak Operator and Everything
> dGRAMOP Development
> Penetration Testing | Animation and Rendering | Web Development | Mobile
> 
> ## App Development | Embedded Electronics | and Misc Java Development.
> 
> http://socnet.16mb.com - My most recent and best project - still buggy,
> still work in progress. Feel free to create an account
> http://dgramop.co - An index page of my projects
> http://m.freeplay.url.ph - A neat looking version of the latter link
> http://freeplay.url.ph - An index page of random experiments.
> http://ppn.hol.es - A currency system that is online that I use to
> "bribe" other children. Over 1.5 years old, I already can name a bunch of
> security flaws. Try ../ attack when creating an account.
> http://m.ppn.hol.es - An even more neat looking version of the above,
> scrolling only works on mobile browsers unless you really zoom in. The junk
> text is there to enable scrolling.

## 

Regards,

## Dhruv Gramopadhye

Controll Freak Operator and Everything
dGRAMOP Development
Penetration Testing | Animation and Rendering | Web Development | Mobile

## App Development | Embedded Electronics | and Misc Java Development.

http://socnet.16mb.com - My most recent and best project - still buggy,
still work in progress. Feel free to create an account
http://dgramop.co - An index page of my projects
http://m.freeplay.url.ph - A neat looking version of the latter link
http://freeplay.url.ph - An index page of random experiments.
http://ppn.hol.es - A currency system that is online that I use to "bribe"
other children. Over 1.5 years old, I already can name a bunch of security
flaws. Try ../ attack when creating an account.
http://m.ppn.hol.es - An even more neat looking version of the above,
scrolling only works on mobile browsers unless you really zoom in. The junk
text is there to enable scrolling.

**Comment by dgramop** [comment]
Can somebody merge this?

File: Wurst Client for MC 1.9.X/src/tk/wurst_client/mods/MassTell.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright © 2014 - 2016 | Wurst-Imperium | All rights reserved.
+ * 
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+ /*
+ * Made by dGRAMOP for WURST!!!
```
Comment (ghost) [inline Wurst Client for MC 1.9.X/src/tk/wurst_client/mods/MassTell.java]: Remove this block comment.

Comment (dgramop) [inline Wurst Client for MC 1.9.X/src/tk/wurst_client/mods/MassTell.java]: This comment is required

Comment (ghost) [inline Wurst Client for MC 1.9.X/src/tk/wurst_client/mods/MassTell.java]: You think so? Let's wait for @Alexander01998 to answer this.

@Alexander01998 wants to add credits like [this](https://github.com/Wurst-Imperium/wurst-client.tk/issues/101).

**Comment by dgramop** [comment]
@Alexander01998 
Done!

File: Wurst Client for MC 1.9.X/src/tk/wurst_client/mods/MassTell.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+			event.cancel();
+			wurst.chat.message("§4§lERROR:§f This server doesn't have TPA.");
+			setEnabled(false);
+		}else if(message.toLowerCase().contains("accepted")
+			&& message.toLowerCase().contains("request")
+			|| message.toLowerCase().contains("akzeptiert")
+			&& message.toLowerCase().contains("anfrage"))
+		{
+			event.cancel();
+			wurst.chat.message("Someone accepted your TPA request. Stopping.");
```
Comment (Alexander01998) [inline Wurst Client for MC 1.9.X/src/tk/wurst_client/mods/MassTell.java]: This does not belong here. Please remove it.

File: Wurst Client for MC 1.9.X/src/tk/wurst_client/mods/MassTell.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+ * 
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+ /*
+ * Use case: 
+ * Regular chat is disabled/you are muted
+ * You want to make it [message] seem less like a broadcast
+ */
```
Comment (Alexander01998) [inline Wurst Client for MC 1.9.X/src/tk/wurst_client/mods/MassTell.java]: Please remove the second block comment.

**Comment by dgramop** [comment]
If it isn't yet evident, I just used the code from MassTPA just to keep things working the same :P

My computer had crashed, after I hit commit and it was still loading, so I guess it was just a small change behind. Sorry! 
@Alexander01998 
I fixed it.

**Comment by liach** [comment]
Where's the implementation of `ChatInputListener`?

**Comment by ghost** [comment]
[Line 32](https://github.com/Wurst-Imperium/Wurst-Client-for-MC-1.9.X/pull/30/files#diff-88ea71eba8fa850253c83b30b7e49a4eR32)?

**Comment by dgramop** [comment]
Hm? Oh yeah. Done. 
@Alexander01998 , mind talkin' a peek?

**Comment by ghost** [comment]
@BrainyBrian It is junk. You had to make a command, not a mod. Even if you made a command, it won't do anything except `/tell YOURNAME`. It would only send a blank message.

**Comment by null-dev** [comment]
Have you actually tested this (because as mccs stated, it doesn't appear to actually send the content of the message)?

It also doesn't even appear to hook into the command manager!

**Comment by ghost** [comment]
@null-dev It is not worth testing.

**Comment by null-dev** [comment]
I know it is probably not worth testing (because it obviously doesn't work)
but I'm trying to tell him that he should at least test any changes he is
trying to get merged so we don't have to yell at him again.

Does Wurst-Client have unit tests even? (It should, there is a lot of math
in here...)

**Comment by liach** [comment]
@mcccs Why did you delete yourself?

**Comment by ghost** [comment]
If you think that I deleted that account, why do you mention that account? How can I hear you?

Alexander banned me, I recreated it but **(I hope) I won't be banned this time, because with this account, I will only answer @mentions** but my last wish, is [my last Pull Request](https://github.com/Wurst-Imperium/Wurst-Client-for-MC-1.9.X/compare/master...mcccs:patch-4) or [this patch](https://gist.github.com/mcccs/7aec5b8f00b65810d0a3cbfc8131c893).

**Comment by ThisTestUser** [comment]
So will you still be active in this account?

**Comment by ghost** [comment]
No

> **I will only answer @mentions**

**Comment by ThisTestUser** [comment]
But will you still logon though?

**Comment by ghost** [comment]
Yeah.

**Comment by dgramop** [comment]
This is kinda been abandoned

On Tue, Jun 28, 2016 at 2:46 PM, mcccs notifications@github.com wrote:

> Yeah.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> https://github.com/Wurst-Imperium/Wurst-Client-for-MC-1.9.X/pull/30#issuecomment-229144607,
> or mute the thread
> https://github.com/notifications/unsubscribe/AKcTuPdFTAvoPmD-rFG_F_Cc-s8XmO6xks5qQWvygaJpZM4IsCuR
> .

## 

Regards,

## Dhruv Gramopadhye

Controll Freak Operator and Everything
dGRAMOP Development
Penetration Testing | Animation and Rendering | Web Development | Mobile

## App Development | Embedded Electronics | and Misc Java Development.

http://socnet.16mb.com - My most recent and best project - still buggy,
still work in progress. Feel free to create an account
http://dgramop.co - An index page of my projects
http://m.freeplay.url.ph - A neat looking version of the latter link
http://freeplay.url.ph - An index page of random experiments.
http://ppn.hol.es - A currency system that is online that I use to "bribe"
other children. Over 1.5 years old, I already can name a bunch of security
flaws. Try ../ attack when creating an account.
http://m.ppn.hol.es - An even more neat looking version of the above,
scrolling only works on mobile browsers unless you really zoom in. The junk
text is there to enable scrolling.

**Comment by IAP-Reloaded** [comment]
Does this work with the Navigator? If not you should make it so

**Comment by dgramop** [comment]
Hold on! I am going to fix this soon.

On Wed, Jun 29, 2016 at 12:21 AM, IAP-Reloaded notifications@github.com
wrote:

> Does this work with the Navigator? If not you should make it so
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> https://github.com/Wurst-Imperium/Wurst-Client-for-MC-1.9.X/pull/30#issuecomment-229252751,
> or mute the thread
> https://github.com/notifications/unsubscribe/AKcTuFFdV5bdgQVa4n6Sdk9eJrCUi1Mjks5qQfLngaJpZM4IsCuR
> .

## 

Regards,

## Dhruv Gramopadhye

Controll Freak Operator and Everything
dGRAMOP Development
Penetration Testing | Animation and Rendering | Web Development | Mobile

## App Development | Embedded Electronics | and Misc Java Development.

http://socnet.16mb.com - My most recent and best project - still buggy,
still work in progress. Feel free to create an account
http://dgramop.co - An index page of my projects
http://m.freeplay.url.ph - A neat looking version of the latter link
http://freeplay.url.ph - An index page of random experiments.
http://ppn.hol.es - A currency system that is online that I use to "bribe"
other children. Over 1.5 years old, I already can name a bunch of security
flaws. Try ../ attack when creating an account.
http://m.ppn.hol.es - An even more neat looking version of the above,
scrolling only works on mobile browsers unless you really zoom in. The junk
text is there to enable scrolling.

**Comment by IAP-Reloaded** [comment]
Okay. You should make it so that you type in the string and then press Enable. Once it mass messages everyone, it will disable itself and you can do it again.

**Comment by vovanre** [comment]
The saga is over?

**Pull request closed without merging**